### PR TITLE
Trap wmstats exception in DBSUploadPoller

### LINKED ISSUE #8764: DBS3Upload crashes if wmstats is unavailable
This is new in the new agent (> 1.1.15.patch2) that relies on wmstats for fetching the parentage map.
I think we can simply wait until the next polling cycle and try again.

Component traceback is
```
2018-08-23 14:40:16,506:140598240118528:INFO:DBSUploadPoller:Starting the DBSUpload Polling Cycle
2018-08-23 14:40:16,524:140598240118528:WARNING:Requests:Http request failed, retrying once again..
2018-08-23 14:40:16,534:140598240118528:WARNING:Service:The cachefile /data/srv/wmagent/v1.1.15.patch2/install/wmagent/DBS3Upload/.wmcore_cache/.wmcore_cache_31961/requests/cmsweb-testbed.cern.ch:8443/15744094587142094_GET_filtered_requests?RequestType=StepChain&ParentageResolved=False&mask=ChainParentageMap does not exist and the service at https://cmsweb-te
stbed.cern.ch:8443/wmstatsserver/data/filtered_requests?RequestType=StepChain&ParentageResolved=False&mask=ChainParentageMap raised a error(111, 'Connection refused') when accessed
2018-08-23 14:40:16,536:140598240118528:ERROR:DBSUploadPoller:Unhandled Exception in DBSUploadPoller!
[Errno 111] Connection refusedTraceback (most recent call last):
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMComponent/DBS3Buffer/DBSUploadPoller.py", line 309, in algorithm
    self.updateDatasetParentageCache()
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMComponent/DBS3Buffer/DBSUploadPoller.py", line 328, in updateDatasetParentageCache
    self.datasetParentageCache = self.wmstatsServerSvc.getChildParentDatasetMap()
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMCore/Services/WMStatsServer/WMStatsServer.py", line 105, in getChildParentDatasetMap
    results = self.getFilteredActiveData(filter, mask)
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMCore/Services/WMStatsServer/WMStatsServer.py", line 92, in getFilteredActiveData
    return self._getResult(callname, verb="GET")
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMCore/Services/WMStatsServer/WMStatsServer.py", line 43, in _getResult
    verb=verb, contentType=contentType)
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMCore/Services/Service.py", line 205, in refreshCache
    self.getData(cachefile, url, inputdata, incoming_headers, encoder, decoder, verb, contentType)
  File "/data/srv/wmagent/v1.1.15.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.15.patch2/lib/python2.7/site-packages/WMCore/Services/Service.py", line 313, in getData
    raise he
error: [Errno 111] Connection refused
```

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #8764

I'm not sure this is the nicest code to put in place, but I chose it instead of handle the exception inside the algorithm method.

Seangchan, once you review I'll continue deploying agents with this patch in

### CODE REVIEW

**Review by ticoann — APPROVED** [review]

**Comment by ticoann** [comment]
Thanks Alan, It is nicer than I was creating. 😄 Patch looks looks good to me.

File: src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py [hunk 1]
```diff
@@ -303,11 +303,12 @@ def algorithm(self, parameters=None):
...
+        logging.info("Starting the DBSUpload Polling Cycle")
+        # refreshing parentageCache every cycle
+        if self.updateDatasetParentageCache() is False:
+            return
+
         try:
-            logging.info("Starting the DBSUpload Polling Cycle")
-            # refreshing parentageCache every cycle
-            self.updateDatasetParentageCache()
-            logging.debug("Dataset parentage map: %s" % self.datasetParentageCache)
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/DBSUploadPoller.py]: Let me add this debug message back in, I removed it by mistake.

**Comment by amaltaro** [comment]
It wasn't catching all the errors it should, like 'Service Unavailable'. It returns an HTTPException object, so we need to look into the `reason` attr. I updated this patch and already tested while Lina was upgrading testbed :-) Now it works!

**Comment by ticoann** [comment]
Thanks Alan, So there are multiple different errors thrown when dbs server is not running?

**Comment by amaltaro** [comment]
Yesterday's exception coincides with a testbed upgrade. I think 'Connection refused' happens when the frontends are still in the DNS cache (but the backend is no longer responsive), while 'Service Unavailable' is the standard message for frontends not accepting any requests anymore.

PS.: replying to GH at 4am?!? Stop it ;)

**Pull request merged**